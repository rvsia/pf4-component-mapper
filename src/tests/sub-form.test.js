import React from 'react';
import toJson from 'enzyme-to-json';
import SubForm from '../form-fields/sub-form';
import { shallow } from 'enzyme';

describe('SubForm component', () => {
  const props = {
    key: 'cosiKey',
    title: 'cosiTitle',
    name: 'cosiName',
    fields: [],
    formOptions: {
      renderForm: (fields, formOptions) => <div>{ 'Here would be form' }</div>,
    },
  };

  it('should render SubForm correctly', () => {
    const wrapper = shallow(
      <SubForm { ...props } />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render SubForm with description correctly', () => {
    const propsDescription = { ...props, description: 'description here!' };
    const wrapper = shallow(
      <SubForm { ...propsDescription } />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render SubForm without title correctly', () => {
    const { key, name, fields, formOptions } = props;
    const propsWithoutTitle = { key, name, fields, formOptions };
    const wrapper = shallow(
      <SubForm { ...propsWithoutTitle } />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
