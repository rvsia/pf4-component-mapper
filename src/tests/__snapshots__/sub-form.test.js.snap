// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubForm component should render SubForm correctly 1`] = `
<Fragment>
  <Title
    className=""
    size="xl"
  >
    cosiTitle
  </Title>
  <Grid
    className=""
    gutter="md"
    lg={null}
    md={null}
    sm={null}
    span={null}
    xl={null}
  >
    <div>
      Here would be form
    </div>
  </Grid>
</Fragment>
`;

exports[`SubForm component should render SubForm with description correctly 1`] = `
<Fragment>
  <Title
    className=""
    size="xl"
  >
    cosiTitle
  </Title>
  <TextContent
    className=""
  >
    <Text
      className=""
      component="small"
      style={
        Object {
          "marginBottom": 0,
        }
      }
    >
      description here!
    </Text>
  </TextContent>
  <Grid
    className=""
    gutter="md"
    lg={null}
    md={null}
    sm={null}
    span={null}
    xl={null}
  >
    <div>
      Here would be form
    </div>
  </Grid>
</Fragment>
`;

exports[`SubForm component should render SubForm without title correctly 1`] = `
<Fragment>
  <Grid
    className=""
    gutter="md"
    lg={null}
    md={null}
    sm={null}
    span={null}
    xl={null}
  >
    <div>
      Here would be form
    </div>
  </Grid>
</Fragment>
`;
